File: code/jvm/service/src/main/java/pt/isel/services/messages/MessageEventService.java
package pt.isel.services.messages;

import pt.isel.domain.messages.UpdatedMessage;
import pt.isel.domain.messages.UpdatedMessageEmitter;

public interface MessageEventService {
    void addEmitter(Long channelId, Long userId, UpdatedMessageEmitter emitter);

    void broadcastMessage(Long channelId, UpdatedMessage signal);
}

File: code/jvm/service/src/main/kotlin/pt/isel/services/messages/MessageEventService.kt
package pt.isel.services.messages

import pt.isel.domain.messages.UpdatedMessage
import pt.isel.domain.messages.UpdatedMessageEmitter

interface MessageEventService {
    fun addEmitter(
        channelId: Long,
        userId: Long,
        emitter: UpdatedMessageEmitter,
    )

    fun broadcastMessage(
        channelId: Long,
        signal: UpdatedMessage,
    )
}
